src/components/shared/Switch/Switch.js

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export const Switch = ({ defaultChecked, disabled, id, name, onChange, text }) => {
+  return (
+    <>
+      <div className="dp-switch">
+        <input
+          type="checkbox"
+          defaultChecked={defaultChecked}
+          disabled={disabled}
+          id={id}
+          name={name}
+          onChange={(e) => onChange(e.target.checked)}
+        />
+        <label htmlFor={id}>
+          <span />
+        </label>
+      </div>
+      <label htmlFor="switch">{text}</label>
+    </>
+  );
+};
+
+Switch.propTypes = {
+  id: PropTypes.string.isRequired,
+  onChange: PropTypes.func.isRequired,
```

Comment:
I'm asking but I'm not sure, can you add some tests validating these?

Reply:
The prop-type warning is printed to the test output, but the test still passes. Jest does not know what to do with the console.error because it is not an Error, and therefore passes the test since it did not throw.